src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java

```diff
@@ -153,8 +154,32 @@ public Map<String, ClinicalDataType> getClinicalAttributeDatatypeMap(StudyViewFi
     )
     @Override
     public List<ClinicalDataCountItem> getClinicalDataCounts(StudyViewFilter studyViewFilter, List<String> filteredAttributes) {
-       return studyViewRepository.getClinicalDataCounts(createContext(studyViewFilter), filteredAttributes);
+
+        var result = studyViewRepository.getClinicalDataCounts(createContext(studyViewFilter), filteredAttributes);
+        
+        // fetch the samples by using the provided study view filter
+        List<Sample> filteredSamples = getFilteredSamples(studyViewFilter);
```

Comment:
I would still suggest this studyViewRepository.getFilteredSamplesCount()

Reply:
@haynescd my understanding was just that you wanted me to call the repository directly, right here